src/_pytest/fixtures.py

```diff
@@ -786,13 +786,21 @@ def fail_fixturefunc(fixturefunc, msg):
 def call_fixture_func(fixturefunc, request, kwargs):
     yieldctx = is_generator(fixturefunc)
     if yieldctx:
-        it = fixturefunc(**kwargs)
-        res = next(it)
-        finalizer = functools.partial(_teardown_yield_fixture, fixturefunc, it)
-        request.addfinalizer(finalizer)
+        generator = fixturefunc(**kwargs)
+        try:
+            fixture_return_value = next(generator)
+        except StopIteration:
+            raise ValueError(
+                "Fixture {} did not yield a value".format(fixturefunc.__name__)
+            )
+        else:
+            finalizer = functools.partial(
+                _teardown_yield_fixture, fixturefunc, generator
+            )
+            request.addfinalizer(finalizer)
```

Comment:
To reduce indentation, would omit the `else` since it's not really needed:

```suggestion
        finalizer = functools.partial(_teardown_yield_fixture, fixturefunc, generator)
        request.addfinalizer(finalizer)
```